src/uct/api/tl.h

```diff
@@ -165,6 +165,12 @@ typedef struct uct_iface_ops {
                                      const uct_device_addr_t *dev_addr,
                                      const uct_ep_addr_t *ep_addr);
 
+    /* endpoint - memory registration */
+
+    ucs_status_t (*ep_mem_reg)(uct_ep_h ep, const uct_iov_t *iov,
+                               size_t iovcnt, uct_md_h *md_p,
```

Comment:
there is no need for this param because an ep is already created on a particular iface which was created on a particular md. so the md should be known to the user up-front.